zebROS_ws/src/tf_object_detection/src/trtdetection.py

```diff
@@ -0,0 +1,249 @@
+        rospack = rospkg.RosPack()
+        THIS_DIR = os.path.join(rospack.get_path('tf_object_detection'), 'src/')
+        PATH_TO_LABELS = os.path.join(THIS_DIR, '2020Game_label_map.pbtxt')
+        rospy.logwarn("Loading labels from " + str(PATH_TO_LABELS))
+        category_index = label_map_util.create_category_index_from_labelmap(PATH_TO_LABELS, use_display_name=True)
+        category_dict = {0: 'background'}
+        for k in category_index.keys():
+            category_dict[k] = category_index[k]['name']
+        
+        viz = BBoxVisualization(category_dict)
+        rospy.logwarn("Obj detection init complete, starting script")
+        
+
+    infrencetime_s = time.time()
+    ori = bridge.imgmsg_to_cv2(msg, "bgr8")
+    # Trying with gpu
+    imgInput = jetson.utils.cudaFromNumpy(ori, isBGR=True)
+    #Normalizes Image between the two values
+    imagerange = (-1., 1.)
+    #Custom function, no documentation
+    jetson.utils.cudaTensorConvert(imgInput, finalgpu, imagerange)
+
+    
+    # This is very slow
+    #np.copyto(host_inputs[0], finalarray.ravel())
+     
+    context.execute_async(bindings=bindings, stream_handle=stream.handle)
+    
+    # host_outputs[1] is not needed
+    # cuda.memcpy_dtoh_async(host_outputs[1], cuda_outputs[1], stream)
+    cuda.memcpy_dtoh_async(host_outputs[0], cuda_outputs[0], stream)
+    stream.synchronize()
+    output = host_outputs[0]
+
+    height, width, channels = ori.shape
+    boxes = []
+    confs = []
+    clss = []
+    detection = TFDetection()
+    detection.header = msg.header
+    detection.header.frame_id = detection.header.frame_id.replace("_optical_frame", "_frame")
+    # Converts numpy array to list becuase extracting indviual items from a list is faster than numpy array
+    # Might be a more optimized way but this takes negligible time 
+    output = output.tolist()
+
+    
+    for i in range(int(len(output) / model.layout)):
+        prefix = i * model.layout
+        
+        conf = float(output[prefix + 2])
+        if conf < min_confidence:
+            continue
+
+        index = int(output[prefix + 0])
+        label = str(output[prefix + 1])
+        #conf = float(output[prefix + 2])
+        xmin = float(output[prefix + 3] * width)
+        ymin = float(output[prefix + 4] * height)
+        xmax = float(output[prefix + 5] * width)
+        ymax = float(output[prefix + 6] * height)
+
+        # ROSifying the code
+        obj = TFObject()
+        obj.confidence = conf
+        obj.tl.x = xmax
+        obj.tl.y = ymax
+        obj.br.x = xmin
+        obj.br.y = ymin
+        obj.id = index
+        obj.label = label
+        detection.objects.append(obj)
+
+        boxes.append([output[prefix + 4], output[prefix + 3], output[prefix + 6], output[prefix + 5]])
+        clss.append(int(output[prefix + 1]))
+        confs.append(output[prefix + 2])
+    
+    pub.publish(detection)
+    
+    # Uncomment for visualization 
```

Comment:
The test_detection script conditially published the debug viz image based on the number of subscribers to the debug topic.  Can you port that over to this code?